STMicroelectronics/STM32L4_L4+/app/az_ulib_dm/inc/az_ulib_registry_api.h

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+
+/**
+ * @file
+ *
+ * @brief   The Registry Editor.
+ *
+ * The Registry Editor is capable of storing key value pairs and stored into flash.
+ */
+
+#ifndef AZ_ULIB_REGISTRY_API_H
+#define AZ_ULIB_REGISTRY_API_H
+
+#include "az_ulib_result.h"
+#include "azure/az_core.h"
+
+#ifndef __cplusplus
+#include <stdint.h>
+#else
+#include <cstdint>
+#endif /* __cplusplus */
+
+#include "azure/core/_az_cfg_prefix.h"
+
+#define MAX_AZ_ULIB_REGISTRY_ENTRIES 50
+#define MAX_AZ_ULIB_REGISTRY_BUFFER 4096
+#define AZ_ULIB_REGISTRY_FLAG_SIZE 8 // in bytes
+
+#define REGISTRY_FREE 0xFFFFFFFFFFFFFFFF
+#define REGISTRY_READY 0x0000000000000000
+#define REGISTRY_DELETED 0x0000000000000000
+
+/**
+ * @brief   Structure for the registry node status.
+ *
+ * Every entry into the registry will have two flags that will indicate the status of the node.
+ * Status can be free, ready to use, and deleted
+ *
+ */
+typedef struct
+{
+  uint64_t ready_flag;
+  uint64_t delete_flag;
+} az_ulib_registry_node_status;
+
+/**
+ * @brief   Structure for the registry data containing key value pair of #az_span pointing to
+ *          location of data in flash.
+ */
+typedef struct
+{
+  /** The #az_span that contains pointer to a location in flash storing the key char string and the
+   * string's size.*/
+  az_span key;
+
+  /** The #az_span that contains pointer to a location in flash storing the valye char string and
+   * the string's size.*/
+  az_span value;
+} az_ulib_registry_node_data;
+
+/**
+ * @brief   Structure for the registry control block
+ *
+ * Every entry into the registry of the device will have a registry control block that contains
+ * relevant information to restore the state of the registry after device power cycle.
+ *
+ */
+typedef struct
+{
+  struct
+  {
+    /** Flags that shows the status of the registry (free, occupied, deleted).*/
```

Comment:
>Flags [](http://example.com/codeflow?start=8&length=5)

"flag." Also I don't think it's a flag if it can take more than 2 values.

Reply:
Each one of them is a flag, but I will clarify the comment. Thanks!